lib/java_buildpack/util/configuration_utils.rb

```diff
@@ -53,7 +64,33 @@ def load(identifier, should_log = true)
 
         CONFIG_DIRECTORY = Pathname.new(File.expand_path('../../../config', File.dirname(__FILE__))).freeze
 
-        private_constant :CONFIG_DIRECTORY
+        ENVIRONMENT_VARIABLE_PATTERN = 'JBP_CONFIG_'
+
+        private_constant :CONFIG_DIRECTORY, :ENVIRONMENT_VARIABLE_PATTERN
+
+        def do_merge(hash_v1, hash_v2, should_log)
+          hash_v2.each do |key, value|
+            if hash_v1.key? key
+              hash_v1[key] = do_resolve_value(key, hash_v1[key], value, should_log)
+            else
+              logger.warn { "User config value for '#{key}' is not valid, existing property not present" } if should_log
+            end
+          end
+          hash_v1
+        end
+
+        def do_resolve_value(key, v1, v2, should_log)
```

Comment:
Do you mean the 'do_resolve_value', method. If I remember right, rubocop will complain about the exclamation marks. The previous lines could be tidied up a little.
```return v2 if !v1_hash && !v2_hash```